C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3312,422 @@ const char *p;
+                                        raiseError("checkSchemaUpdateRequired", "The property return must be an array");
+                                        return false;
+                                }
+
+				for (auto& v : indexes.GetArray())
+                                {
+                                        if (v.IsObject())
+                                        {
+                                                if (v.HasMember("index"))
+                                                {
+                                                        if (!v["index"].IsArray())
+                                                        {
+                                                                Logger::getLogger()->error("%s %d ", __FUNCTION__, __LINE__, "extracting index values , expecting an array here");
+								return false;
+                                                        }
+                                                        else
+                                                        {
+								std::string s;
+								for (auto& i : v["index"].GetArray())
+								{
+									s.append(i.GetString());
+									s.append(",");
+								}
+								// remove last comma
+								if ( s[s.size()-1] == ',')
+								{
+									s.erase(s.size()-1);
+								}
+
+                                                               	indexSet.insert(s);
+                                                        }
+                                                }
+                                        }
+                                }
+
+				tableColumnMap[name] = columnSet;
+				tableIndexMap[name] = indexSet;
+
+			}
+		}
+
+	}
+
+	return true;
+}
+/**
+ * Create scema of tables
+ *
+ * @param payload   The  payload containing information about schema of 
+ *                  tables to create
+ * @return true if the tables can be crated successfully
+ */
+int Connection::create_schema(std::string payload)
+{
+	Logger::getLogger()->error ( "%s:%d payload is  %s", __FUNCTION__, __LINE__, payload.c_str());
+
+	Document document;
+	std::string schema;
+	int version;
+	const char *logSection="CreatingSchema";
+	unsigned long rowsAffectedLastCommand = 0;
+	std::unordered_map<std::string, unordered_set<std::string> > columnMap;
+        std::unordered_map<std::string, unordered_set<std::string> > indexMap;
+
+	bool schemaUpdateRequired = false;
+
+	try {
+                if (payload.empty())
+                {
+                        return -1;
+                }
+                else
+                {
+                        if (document.Parse(payload.c_str()).HasParseError())
+                        {
+                                raiseError("retrieve", "Failed to parse JSON payload");
+                                return -1;
+                        }
+			if (!document.HasMember("schema"))
+                        {
+				Logger::getLogger()->error("schema absent from payload, so exiting ");
```

Comment:
exciting is a scary thing to log, it makes it sound as if the storage service is going to terminate!